packages/shell-api/src/helpers.ts

```diff
@@ -487,33 +498,45 @@ export async function getPrintableShardStatus(
   ]);
   result.balancer = balancerRes;
 
-  const databases = await (await configDB.getCollection('databases').find())
-    .sort({ name: 1 })
-    .toArray();
-
+  // All databases in config.databases + those implicitly referenced
+  // by a sharded collection in config.collections
+  // (could become a single pipeline using $unionWith when we drop 4.2 server support)
+  const [databases, collections] = await Promise.all([
+    (async () =>
+      await (await configDB.getCollection('databases').find())
+        .sort({ _id: 1 })
+        .toArray())(),
+    (async () =>
+      await (
+        await configDB.getCollection('collections').find({
+          ...onlyShardedCollectionsInConfigFilter,
+        })
+      )
+        .sort({ _id: 1 })
+        .toArray())(),
+  ]);
```

Comment:
This isn't directly related to the original change, but it seems that sharding collections doesn't necessarily introduce entries in the `config.databases` collection, which makes sense given that sharding doesn't explicitly need to be enabled for databases anymore since https://jira.mongodb.org/browse/SERVER-60926